src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs

```diff
@@ -15,6 +15,11 @@ public void ConnectTo(string url)
         {
             Url = url;
 
+            if (url == null)
```

Comment:
Don't think so. This is called on application start as well which means if an exception is thrown the app will crash.